e3db/tests/test_integration.py

```diff
@@ -889,6 +909,111 @@ def test_read_write_config(self):
+        client_name = "client_no_signing_{0}".format(binascii.hexlify(os.urandom(16)))
+        test_client = e3db.Client.register(token, client_name, client_public_key, api_url=api_url)
+        client_api_key_id = test_client.api_key_id
+        client_api_secret = test_client.api_secret
+        client_id = test_client.client_id
+        client_no_signing_config = e3db.Config(
+            client_id,
+            client_api_key_id,
+            client_api_secret,
+            client_public_key,
+            client_private_key,
+            api_url=api_url
+        )
+        client_no_signing_keys = e3db.Client(client_no_signing_config())
+
+        client_public_signing_key, client_private_signing_key = e3db.Client.generate_signing_keypair()
+        client_signing_config = e3db.Config(
+            client_id,
+            client_api_key_id,
+            client_api_secret,
+            client_public_key,
+            client_private_key,
+            api_url=api_url,
+            public_signing_key=client_public_signing_key,
+            private_signing_key=client_private_signing_key
+        )
+        client_signing_keys = e3db.Client(client_signing_config())
+
+        assert(client_no_signing_keys.private_signing_key == "")
+        assert(client_no_signing_keys.public_signing_key == "")
+        assert(client_signing_keys.private_signing_key == client_private_signing_key)
+        assert(client_signing_keys.public_signing_key == client_public_signing_key)
+
+    def test_read_config_no_signing_keys(self):
+        """
+        Register a client without signing keys with a registration token.
+        Read config from storage to instantiate a client showing config 
+        without signing keys is able to be used in client instantiation. 
+        """
+
+        config_client_public_key, config_client_private_key = e3db.Client.generate_keypair()
+        config_client_name = "client_{0}".format(binascii.hexlify(os.urandom(16)))
+        test_config_client = e3db.Client.register(token, config_client_name, config_client_public_key, api_url=api_url)
+        config_client_api_key_id = test_config_client.api_key_id
+        config_client_api_secret = test_config_client.api_secret
+        config_client_id = test_config_client.client_id
+
+        config_client_config = e3db.Config(
+            config_client_id,
+            config_client_api_key_id,
+            config_client_api_secret,
+            config_client_public_key,
+            config_client_private_key,
+            api_url=api_url
+        )
+        config_name = "integration_config_no_signing{0}".format(binascii.hexlify(os.urandom(16)))
+        # write config with profile 'config_name' ~/.tozny/<profile>/e3db.json
+        config_client_config.write(config_name)
+
+        # Remove signing keys from config. By default Config.write() writes signing keys, including
+        # empty strings if none were provided. 
+        home = os.path.expanduser('~')
```

Comment:
The concern is not so much reading the config from the file. You don't need all this complexity of modifying the file. One line 1012 you can just use read_config_profile.pop("public_signing_key", None) https://docs.python.org/3/library/stdtypes.html#dict.pop to remove the values. from there we can validate that the client can be instantiated.

Reply:
Thanks! I was concerned I was doing way too much to remove the keys. Will update

Reply:
Updated, re-ran test and it passes.